### PR TITLE
Automatically order units away when trying to deploy transform

### PR DESCRIPTION
**Pull request by Mailaender**
This expands on https://github.com/OpenRA/OpenRA/pull/9239.

### CODE REVIEW

**Review by RoosterDragon — APPROVED** [review]

File: OpenRA.Mods.Common/Traits/Transforms.cs [hunk 1]
```diff
@@ -97,6 +98,40 @@ public bool CanDeploy()
 			return buildingInfo == null || self.World.CanPlaceBuilding(self.Location + Info.Offset, actorInfo, buildingInfo, self);
 		}
 
+		IEnumerable<Order> ClearBlockersOrders(World world, CPos topLeft)
+		{
+			var allTiles = buildingInfo.Tiles(topLeft).ToArray();
```
Comment (PunkPun) [inline OpenRA.Mods.Common/Traits/Transforms.cs]:
```suggestion
			var allTiles = buildingInfo.Tiles(topLeft).ToList();
```

Not that it matters but a ToList is faster here

**Review by PunkPun — COMMENTED** [review]
The original function is basically identical, and it would be identical if one were to make them static. Could we perhaps reuse the logic? They are in the same scope

File: OpenRA.Mods.Common/Traits/Transforms.cs [hunk 2]
```diff
@@ -97,6 +98,20 @@ public bool CanDeploy()
...
+		IEnumerable<Order> ClearBlockersOrders(World world, CPos topLeft)
+		{
+			var allTiles = buildingInfo.Tiles(topLeft).ToList();
+			var adjacentTiles = Util.ExpandFootprint(allTiles, true).Except(allTiles)
+				.Where(world.Map.Contains).ToList();
+
+			var blockers = allTiles.SelectMany(world.ActorMap.GetActorsAt)
+				.Where(a => a.Owner == self.Owner && a != self && a.IsIdle)
+				.Select(a => new TraitPair<IMove>(a, a.TraitOrDefault<IMove>()))
+				.Where(x => x.Trait != null);
```
Comment (PunkPun) [inline OpenRA.Mods.Common/Traits/Transforms.cs]: I meant including these as well. All you need to pass in is world, buildingInfo, and owner

Comment (Mailaender) [reply OpenRA.Mods.Common/Traits/Transforms.cs]: No, the required parameters are many more, which is why I purposefully avoided this complicated way, but here you go.

**Review by PunkPun — APPROVED** [review]

**Comment by PunkPun** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/72d243da1e434a57441640828c93ed32996c52e2...388c703b9a94c5c242f8b30247d725d21f5ee118)

**Pull request merged**